documentation/src/main/asciidoc/search-dsl.asciidoc

```diff
@@ -230,24 +360,168 @@ include::todo-placeholder.asciidoc[]
+----
+<1> Build the query as usual.
+<2> Retrieve a `SearchQuery` object.
+====
+
+This query object supports all <<search-dsl-query-fetching-results,`fetch*` methods supported by the query DSL>>.
+The main advantage over calling these methods directly at the end of a query definition
+is mostly related to debugging (see <<search-dsl-query-debugging>>),
+but the query object can also be useful if you need an adapter to another API.
+
+Hibernate Search provides an adapter to JPA and Hibernate ORM's native APIs,
+i.e. a way to turn a `SearchQuery` into a `javax.persistence.TypedQuery` (JPA)
+or a `org.hibernate.query.Query` (native ORM API):
+
+.Getting a `SearchQuery` object
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/searchdsl/query/QueryDslIT.java[tags=searchQuery-toORM]
+----
+<1> Build the query as usual.
+<2> Retrieve a `SearchQuery` object.
+<3> Turn the `SearchQuery` object into a JPA query.
+<4> Turn the `SearchQuery` object into a Hibernate ORM query.
+====
+
+// Search 5 anchors backward compatibility
+[[_resulttransformer]]
+[WARNING]
+====
+The resulting query *does not support all operations*,
+so is recommended to only convert search queries when absolutely required,
+for example when integrating with code that only works with Hibernate ORM queries.
 
-// TODO [[_resulttransformer]] https://docs.jboss.org/hibernate/search/5.11/reference/en-US/html_single/#_resulttransformer
+The following operations are expected to work correctly in most cases,
+even though they may behave slightly differently from what is expected from a JPA `TypedQuery`
+or Hibernate ORM `Query` in some cases
+(including, but not limited to, the type of thrown exceptions):
+
+* Hit retrieval methods (`list`, `getResultList`, `uniqueResult`, ... ).
+* `setFirstResult`/`setMaxResults` and getters.
+* `setFetchSize`
+* `unwrap`
+
+The following operations are known not to work correctly,
+with no plan to fix them at the moment:
+
+* Hints (`setHint`, ...).
+* Parameter-related methods (`setParameter`, ...).
+* Result transformer (`setResultTransformer`, ...);
+use <<search-dsl-projection-composite,composite projections>> instead.
+* Lock-related methods (`setLockOptions`, ...).
+* And more (this list is not exhaustive).
+====
 
 [[search-dsl-query-debugging]]
 === Debugging a query
+
+[[search-dsl-query-debugging-matches]]
+==== Explaining matches
+
+When some documents unexpectedly match or don't match,
+you will need information about the exact query being executed,
+and about the index content.
+
+To gain insight about what ends up being executed exactly,
+one option is to <<search-dsl-query-object,create a `SearchQuery` object>>
+using `toQuery()` at the end of the query definition,
+then call `toString()` to get a String representation of that query.
+
+Another option is to take advantage of logs:
+all executed search queries are logged to the log category `org.hibernate.search.fulltext_query`
```

Comment:
I have a doubt here: is the dash after `fulltext` correct?

Reply:
You're right, I was using the log category from Search 5. In Search 6, it's simply `org.hibernate.search.query`.